lib/ship.rb

```diff
@@ -38,36 +38,34 @@ def build
 
   def destroy
     @location.each { |xy| @matrix[xy[0]][xy[1]] = ' ' }
-    @location = []
+    @location = Array.new
   end
 
   def save(xy)
-    @location.push(xy)
+    @location.push xy
     @matrix[xy[0]][xy[1]] = true
   end
 
   # returns valid surrounding mask
   def take_mask(xy, exception = nil)
-    return [] unless xy
-    x = xy[0]
+    return Array.new unless xy
+    x = xy.first
```

Comment:
I would keep previous version because the context is more readable like: `xy[0], xy[1], ...`